### PR TITLE
Format codes with clang-format

### PR DESCRIPTION
**Pull request by alesiong**
also refactor some code with cpplinter

### CODE REVIEW

**Comment by alesiong** [comment]
@Yexiaoxing 
cpplinter reports this
```
src/include/json/Json.h(20):  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
src/include/json/Json.h(21):  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
src/include/json/Json.h(38):  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
src/include/json/Json.h(39):  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
src/include/json/Json.h(166):  Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]
src/include/json/Json.h(233):  Use int16/int64/etc, rather than the C type long  [runtime/int] [4]
src/include/json/Json.h(234):  Use int16/int64/etc, rather than the C type long  [runtime/int] [4]
src/include/json/Json.h(248):  Use int16/int64/etc, rather than the C type long  [runtime/int] [4]
src/include/json/Json.h(254):  Use int16/int64/etc, rather than the C type long  [runtime/int] [4]
```
So do these constructors need to be implicit? And can `long` be changed to `int64_t` (as on windows x64, `long` is 32 bit)

File: src/base64/base64_functions.cpp [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 #include <iomanip>
 #include <sstream>
 #include <string>
-#include <bitset>
-#include "base64/base64_functions.h"
 
 namespace cuhksz {
-static const std::string codes = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
+static const char codes[] =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=";
```
Comment (Yexiaoxing) [inline src/base64/base64_functions.cpp]: necessary?

File: src/graph/Point.cpp [hunk 2]
```diff
@@ -1,8 +1,3 @@
 #include "graph/Point.h"
 
-namespace cuhksz {
-
-
-
-
-}	// namespace cuhksz
+namespace cuhksz {}  // namespace cuhksz
```
Comment (Yexiaoxing) [inline src/graph/Point.cpp]: remove empty cpp if not required

File: src/include/geometry/Circle.h [hunk 3]
```diff
@@ -4,12 +4,9 @@
 namespace cuhksz {
 
 class Circle {
-private:
-
+ private:
 };
 
+}  // namespace cuhksz
```
Comment (Yexiaoxing) [inline src/include/geometry/Circle.h]: not implemented yet?

File: src/base64/base64_functions.cpp (hunk 1 above)
Comment (alesiong) [inline src/base64/base64_functions.cpp]: This modification can save the static memory space and remove the needs of exit-time destructing

File: src/include/geometry/Circle.h (hunk 3 above)
Comment (alesiong) [inline src/include/geometry/Circle.h]: It should be implemented by @37790928

File: src/json/Json.cpp [hunk 4]
```diff
@@ -131,16 +148,17 @@ JSONObject parse_number(const std::string &str, size_t &offset) {
   std::string val, exp_str;
   char c;
   bool isDouble = false;
-  long exp = 0;
+  int exp = 0;
```
Comment (alesiong) [inline src/json/Json.cpp]: @Yexiaoxing I think there's no problem to change this `long` to `int`

**Pull request merged**